src/BuildingRegistry/Building/BuildingUnit.cs

```diff
@@ -432,6 +432,40 @@ public void DetachAddressBecauseAddressWasRemoved(AddressPersistentLocalId addre
                 newAddressPersistentLocalId));
         }
 
+        public void Demolish()
+        {
+            if (IsRemoved)
+            {
+                return;
+            }
+
+            if (Status == BuildingUnitStatus.NotRealized || Status == BuildingUnitStatus.Retired)
```

Comment:
Do we have a guard status?

Reply:
A guard will throw an exception. Here we don't want to throw an exception, but rather do nothing. So this seems ok.